### PR TITLE
test(d4): add unit tests for Context Extraction regex patterns

### LINKED ISSUE #3890: Add unit tests for D-4 Context Extraction regex patterns
## Background

PR #3889 improved the file path extraction regex patterns to support Python SyntaxError format. While the patterns have been manually tested, we should add unit tests to ensure they work correctly and prevent regressions.

## Scope

Add unit tests for:
- `_extract_file_path_from_error()` - single file extraction
- `_extract_file_paths_from_error()` - multi-file extraction

### Pattern 1: Lint format (flake8, pylint, eslint)
- `path/to/file.py:123:45: E501 line too long`
- `src/main.js:10:5: error`

### Pattern 2: Compiler format
- `path/to/file.py(123): error`

### Pattern 3: Error in / File format (with optional quotes)
- `Error in path/to/file.py`
- `File path/to/file.py`
- `File "path/to/file.py"` (quoted)
- `in file src/main.js`

### Pattern 4: Python SyntaxError format
- `File "path/to/file.py", line 10`
- `  File "/usr/lib/python3.10/ast.py", line 50, in parse`
- Multi-line traceback with multiple File references

## Acceptance Criteria

- [ ] Unit tests cover all 4 patterns
- [ ] Tests verify correct extraction (no extra quotes, correct path)
- [ ] Tests verify no false positives
- [ ] Tests added to `orchestrator/tests/` directory

## Related

- PR #3889: feat(d4): improve Context Extraction for Python SyntaxError format
- Issue #3890: D-4 Context Extraction improvements
- Blueprint: EPIC D - Autonomous Coder Agent Family

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Closes #3890. Adds comprehensive unit tests for the D-4 Context Extraction regex patterns that were improved in PR #3889.

**New tests for `_extract_file_path_from_error()`** (5 tests):
- Python SyntaxError format: `File "path/file.py", line N`
- Nested paths in SyntaxError
- Python tracebacks with multiple File entries
- Quoted file paths
- Case-insensitive File keyword matching

**New tests for `_extract_file_paths_from_error()`** (13 tests):
- Single and multiple file extraction (flake8, compiler formats)
- Deduplication of repeated file paths
- Mixed error formats in same output
- Python SyntaxError and traceback formats
- Edge cases: empty input, no matches, max files limit
- Various file extensions (.py, .js, .ts, .go, .rs)

Also adds `general_coder_max_files = 5` to `FakeSettings` for test compatibility.

## Updates since last revision

- Changed `test_limits_to_max_files` assertion from `len(result) <= 5` to `len(result) == 5` for stricter validation (addresses Gemini Code Assist review feedback)

## Review & Testing Checklist for Human

- [ ] Run the full test suite locally: `cd handoff/20250928/40_App/orchestrator && pytest tests/test_fixer_node_unit.py -v -k "TestExtractFilePathFromError or TestExtractFilePathsFromError"`
- [ ] Verify all 28 tests pass (15 for single-file extraction, 13 for multi-file extraction)

### Notes

Requested by: @RC918
Link to Devin run: https://app.devin.ai/sessions/16834c21998741ca99953fee3b80910f

### CODE REVIEW

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

*Note: 35 comments for 14 files not in this PR diff were filtered to reduce noise from pre-existing issues.*

### File-Level Comments

**General** (high)
## Discovery Audit: Silent Failure Detected

The following test files were added/modified in this PR but were **not executed** in CI:

- `tests/test_fixer_node_unit.py`

### Possible Causes
1. Test directory not listed in `pytest.ini` testpaths
2. Test file naming doesn't match `test_*.py` pattern
3. Test collection error (check CI logs for import errors)

### How to Fix
1. Verify the test directory is in `pytest.ini` under `testpaths`
2. Check CI logs for collection errors
3. Run `pytest --collect-only` locally to verify discovery

---
*This check is part of [Discovery 全鏈路治理](https://github.com/RC918/morningai/issues/3310)*

File: handoff/20250928/40_App/orchestrator/tests/test_fixer_node_unit.py [hunk 1]
```diff
@@ -1050,3 +1052,157 @@ def test_handles_various_extensions(self):
...
+        error = """src/file1.py:1:1: error
+src/file2.py:1:1: error
+src/file3.py:1:1: error
+src/file4.py:1:1: error
+src/file5.py:1:1: error
+src/file6.py:1:1: error
+src/file7.py:1:1: error"""
+        result = _extract_file_paths_from_error(error)
+        # Should be limited to 5 files (general_coder_max_files default)
+        assert len(result) <= 5
```
[the remark this answers is not in the document]
Comment (RC918) [reply handoff/20250928/40_App/orchestrator/tests/test_fixer_node_unit.py]: Valid suggestion - implemented in dffc3aae. Changed `assert len(result) <= 5` to `assert len(result) == 5` for stricter validation.

Regarding the first suggestion about adding `len()` assertions before `set()` comparisons: This is a valid improvement for robustness, but the current tests already verify correctness. The `set()` comparison ensures all expected files are present, and tests like `test_extracts_duplicate_files_once` specifically verify deduplication behavior. I'll leave this as a minor follow-up improvement.

**Comment by RC918** [comment]
**評估結果：False Positive**

這是一個誤報。測試已在本地成功執行並通過（28 個測試全部通過）。

CI 可能有不同的 test discovery 配置，但這不是 PR 的問題 - 這是一個獨立於此 PR 的 CI 配置問題。測試檔案命名符合 `test_*.py` 模式，且位於正確的 `tests/` 目錄中。

本地測試執行結果：
```
tests/test_fixer_node_unit.py ............................               [100%]
================ 28 passed, 26 deselected, 6 warnings in 1.56s =================
```

File: handoff/20250928/40_App/orchestrator/tests/test_fixer_node_unit.py [hunk 2]
```diff
@@ -1050,3 +1052,157 @@ def test_handles_various_extensions(self):
...
+        """Test that results are limited to max files (default 5)."""
+        error = """src/file1.py:1:1: error
+src/file2.py:1:1: error
+src/file3.py:1:1: error
+src/file4.py:1:1: error
+src/file5.py:1:1: error
+src/file6.py:1:1: error
+src/file7.py:1:1: error"""
+        result = _extract_file_paths_from_error(error)
+        # Should be limited to 5 files (general_coder_max_files default)
```
[the remark this answers is not in the document]
Comment (RC918) [reply handoff/20250928/40_App/orchestrator/tests/test_fixer_node_unit.py]: **評估結果：Invalid**

這個建議是無效的。測試使用 `FakeSettings` 中定義的 `general_coder_max_files = 5`，而不是 production 的預設值。測試註解中的 "default 5" 指的是 `FakeSettings` 中的值，這是正確的。

如果 production 的預設值改變，測試仍然會正常運作，因為它使用的是 `FakeSettings` 中的固定值。這是測試隔離的正確做法。
